scripts/prettier.js

```diff
@@ -55,4 +28,67 @@ async function runPrettierForAll() {
   });
 }
 
-runPrettierForAll();
+function parseArgs() {
+  return require('yargs')
+    .usage('Usage: prettier [commitHash] [options]')
+    .example('prettier', 'run prettier only on changed files')
+    .example('prettier HEAD~3', 'run prettier only on changed files since HEAD~3')
+    .options({
+      all: {
+        description: 'Run prettier on all projects',
+        boolean: true,
+      },
+    })
+    .alias('h', 'help').argv;
+}
+
+/**
+ *
+ * @param {PQueue} queue
+ */
+async function runOnChanged(queue) {
+  const prettierIntroductionCommit = 'HEAD~1';
+  const passedDiffTarget = parsedArgs._.length ? parsedArgs._[0] : prettierIntroductionCommit;
+
+  const projectRootPath = path.resolve(__dirname, '..');
+  const cmd = `git --no-pager diff ${passedDiffTarget} --diff-filter=AM --name-only --stat-name-width=0`;
+
+  const gitDiffOutput = execSync(cmd, { cwd: projectRootPath });
+  const prettierExtRegex = new RegExp(`\\.(${prettierExtensions.join('|')})$`);
+  const files = gitDiffOutput
+    .toString('utf8')
+    .split(os.EOL)
+    .filter(fileName => prettierExtRegex.test(fileName));
+
+  const fileGroups = [];
+  for (let chunkStart = 0; chunkStart < files.length; chunkStart += numberOfCpus) {
+    fileGroups.push(files.slice(chunkStart, chunkStart + numberOfCpus));
+  }
+
+  console.log(files);
+  process.exit(1);
+
+  await queue.addAll(
+    fileGroups.map(group => () => {
+      console.log(`Running for ${group.length} files!`);
+      runPrettier(group, true /*async*/);
+    }),
+  );
+}
+
+/**
+ *
+ * @param {PQueue} queue
+ */
+async function runOnAll(queue) {
+  // Run on groups of files so that the operations can run in parallel
+  const root = findGitRoot();
+  await queue.add(() => runPrettierForFolder(root, true, true));
+  await queue.addAll(
+    ['apps', 'packages/!(fluentui)', 'packages/fluentui', '{.*,scripts,typings}'].map(name => () =>
+      runPrettierForFolder(name),
+    ),
+  );
+}
+
+main();
```

Comment:
To avoid potential unhandled promise rejection warnings, might be good to do something like this:
```suggestion
main().catch(err => {
  console.error(err);
  process.exit(1);
});
```

Reply:
good idea, although this wasn't handled previously, nor is it tested.

Reply:
btw, don't we have an eslint rule for this ? AFAIR it's shipped with standard typescript-eslint plugin

Reply:
When we migrated from tslint to eslint we didn't initially set it up to run on JS files, or actually anything besides files under `<package>/src`, probably to make the migration slightly less complex (IIRC we only ran tslint on `<package>/src`). Linting for scripts and JS files would be good to add, we just haven't done it yet.